calendar/experiments/calendar/parent/ext-calendar-provider.js

```diff
@@ -409,6 +437,19 @@ this.calendar_provider = class extends ExtensionAPI {
               };
             },
           }).api(),
+
+          onFreeBusy: new EventManager({
+            context,
+            name: "calendar.provider.onFreeBusy",
+            register: fire => {
+              let provider = new ExtFreeBusyProvider(fire);
+              cal.getFreeBusyService().addProvider(provider);
+
+              return () => {
+                cal.getFreeBusyService().removeProvider(provider);
```

Comment:
Can you confirm that the freebusy provider gets removed on shutdown as well? I'm guessing all the undo functions are called, but would be great to check.

Reply:
The provider gets removed when the listener does, either through the extension explicitly removing it, or when the extension is disabled, either temporarily or via uninstallation or shutdown.